packages/composite-checkout/src/components/checkout-steps.js

```diff
@@ -107,21 +111,23 @@ export function CheckoutSteps( { children } ) {
 }
 
 export function CheckoutStep( {
-	children,
-	title,
+	activeStepContent,
+	titleContent,
 	stepId,
-	isComplete,
+	isCompleteCallback,
 	stepNumber,
 	nextStepNumber,
 	isStepActive,
 	isStepComplete,
 	setActiveStepNumber,
 	setStepCompleteStatus,
 } ) {
+	const localize = useLocalize();
+	const { totalSteps } = useContext( CheckoutStepDataContext );
 	const { setFormPending, setFormReady } = useFormStatus();
 	const goToThisStep = () => setActiveStepNumber( stepNumber );
 	const goToNextStep = () => {
-		const completeResult = isComplete();
+		const completeResult = isCompleteCallback();
 		if ( completeResult.then ) {
```

Comment:
Is this a reliable way to detect whether something is a Promise? (I really don't know)

Reply:
I actually did some research, and this is the _only_ proper way to detect if something is a promise! More specifically, we are checking if something is "thennable".